src/pull_module/libgit2.cpp

```diff
@@ -200,17 +202,85 @@ HfDownloader::HfDownloader() {
     this->hfEndpoint = "";
     this->hfToken = "";
     this->httpProxy = "";
+    this->overwritePath = false;
+}
+
+std::string HfDownloader::getGraphDirectory() {
+    return this->downloadPath;
+}
+
+std::string HfDownloader::getGraphDirectory(const std::string& inDownloadPath, const std::string& inSourceModel) {
+    std::string fullPath = FileSystem::joinPath({inDownloadPath, inSourceModel});
+    // Windows path creation
+    if (FileSystem::getOsSeparator() != "/") {
```

Comment:
I think we should free HfDownloader from worrying if path separator is correct here - it should get proper filepaht os::filepath maybe instead of string?

Reply:
This string is from the user thus we want to make sure we have correct paths for folder creation.

Reply:
This tells me that we have a bug in FileSystem::joinPath -> we create path that is not OS agnostic. To be fixed there.